src/main/kotlin/io/github/nomisrev/repo/ArticlePersistence.kt

```diff
@@ -23,6 +29,8 @@ interface ArticlePersistence {
 
   /** Verifies if a certain slug already exists or not */
   suspend fun exists(slug: Slug): Boolean
+
+  suspend fun getArticleBySlug(slug: Slug): Either<DomainError, Articles>
```

Comment:
We should prefer more precise errors if we can. Variance will allows us to automatically lift to supertypes where needed.

```suggestion
  suspend fun getArticleBySlug(slug: Slug): Either<ArticleBySlugNotFound, Articles>
```